lightwood/helpers/general.py

```diff
@@ -169,17 +167,19 @@ def evaluate_array_r2_accuracy(
     """
     base_acc_fn = kwargs.get('base_acc_fn', lambda t, p: max(0, r2_score(t, p)))
 
-    aggregate = 0.0
+    nan_mask = (~np.isnan(true_values)).astype(int)
+    predictions *= nan_mask
+    true_values = np.nan_to_num(true_values, 0.0)
 
-    fh = 1 if not isinstance(predictions[0], list) else len(predictions[0])
-    if fh == 1:
-        predictions = [[p] for p in predictions]
+    fh = kwargs.get('ts_analysis', {}).get('tss', None)
+    fh = fh.nr_predictions if fh is not None else 1
 
-    # only evaluate accuracy for rows with complete historical context
-    if kwargs.get('ts_analysis', {}).get('tss', False):
+    if kwargs.get('ts_analysis', {}).get('tss', False) and not kwargs['ts_analysis']['tss'].eval_cold_start:
```

Comment:
Why is `tss` part of something called `ts_analysis` here? Couldn't we be passing this directly?

Reply:
No, because under certain conditions `ts_analysis` also includes group information and some stats that might be used inside accuracy functions like MASE.